File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/Order.cpp
#include <vector>
#include <unordered_map>
#include <string>

struct Dish {
    std::string name;
    double price;
    int count;
};

class Order {
private:
    std::vector<Dish> menu;
    std::vector<Dish> selected_dishes;
    std::unordered_map<std::string, double> sales;

public:
    Order() {}

    bool add_dish(Dish dish) {
        for (auto& menu_dish : menu) {
            if (dish.name == menu_dish.name) {
                if (menu_dish.count < dish.count) {
                    return false;
                } else {
                    menu_dish.count -= dish.count;
                    break;
                }
            }
        }
        selected_dishes.push_back(dish);
        return true;
    }

    double calculate_total() {
        double total = 0;
        for (const auto& dish : selected_dishes) {
            total += dish.price * dish.count * sales[dish.name];
        }
        return total;
    }

    double checkout() {
        if (selected_dishes.empty()) {
            return -1.0;
        }
        double total = calculate_total();
        selected_dishes.clear();
        return total;
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/Order.py
class Dish:
    def __init__(self, dish="", price=0.0, count=0):
        self.dish = dish
        self.price = price
        self.count = count


class Order:
    def __init__(self):
        self.menu = []
        self.selected_dishes = []
        self.sales = {}

    def add_dish(self, dish):
        for menu_dish in self.menu:
            if dish.dish == menu_dish.dish:
                if menu_dish.count < dish.count:
                    return False
                else:
                    menu_dish.count -= dish.count
                    break
        # Copy semantics: C++ push_back copies the dish
        self.selected_dishes.append(Dish(dish.dish, dish.price, dish.count))
        return True

    def calculate_total(self):
        total = 0.0
        for dish in self.selected_dishes:
            if dish.dish in self.sales:
                total += dish.price * dish.count * self.sales[dish.dish]
        return total

    def checkout(self):
        if not self.selected_dishes:
            return 0
        total = self.calculate_total()
        self.selected_dishes.clear()
        return total